### PR TITLE
Fix handling of consumable item fluid containers.

### LINKED ISSUE #4539: [1.11.2] Placing water in a machine with a forestry tin can or capsule causes a ticking tile entity
Using a forestry tin can or capsule instead of a bucket in Mekanism machines causes ticking block crash.

1. Place electrolytic separator
2. Put water can in water slot
3. Crash

Mekanism 307
Forge 13.20.1.2386
Minecraft 1.11.2

Log:
https://paste.ee/p/oxUq9

### PR DESCRIPTION
**Pull request by thiakil**
Fixes #4539.

Basically Tom's changes, plus changed to passing around the FluidHandler instead of the itemstack.

Tested with Buckets, Forestry cans & mek tanks.

### CODE REVIEW

File: src/main/java/mekanism/common/util/FluidContainerUtils.java [hunk 1]
```diff
@@ -27,40 +27,46 @@ public static FluidStack extractFluid(FluidTank tileTank, TileEntityContainerBlo
...
 	{
-		IFluidHandlerItem handler = FluidUtil.getFluidHandler(container);
-		
-		if(handler == null || FluidUtil.getFluidContained(container) == null)
+		if (handler == null)
 		{
 			return null;
 		}
+
+		FluidStack fluidStack = handler.drain(Integer.MAX_VALUE, false);
```
Comment (thiakil) [inline src/main/java/mekanism/common/util/FluidContainerUtils.java]: NB: this is essentially what `FluidUtil.getFluidContained` does

File: src/main/java/mekanism/common/util/FluidContainerUtils.java [hunk 2]
```diff
@@ -27,40 +27,46 @@ public static FluidStack extractFluid(FluidTank tileTank, TileEntityContainerBlo
...
 	public static FluidStack extractFluid(FluidTank tileTank, TileEntityContainerBlock tile, int slotID, FluidChecker checker)
 	{
-		FluidStack ret = extractFluid(tileTank.getCapacity()-tileTank.getFluidAmount(), tile.inventory.get(slotID), checker);
-		tile.inventory.set(slotID, FluidUtil.getFluidHandler(tile.inventory.get(slotID)).getContainer());
+		IFluidHandlerItem handler = FluidUtil.getFluidHandler(tile.inventory.get(slotID));
+		FluidStack ret = null;
+		if (handler != null)
+		{
+			ret = extractFluid(tileTank.getCapacity() - tileTank.getFluidAmount(), handler, checker);
+			tile.inventory.set(slotID, handler.getContainer());
```
Comment (thommy101) [inline src/main/java/mekanism/common/util/FluidContainerUtils.java]: How does this behave with the Forestry cans? Do you get empty ones or none?

Comment (thiakil) [inline src/main/java/mekanism/common/util/FluidContainerUtils.java]: None - the same result as with the bottler as far as I could tell

**Comment by aidancbrady** [comment]
Thanks!

**Pull request merged**